src/Sarif/NotYetGenerated/SarifRewritingVisitor.cs

```diff
@@ -684,7 +726,14 @@ public virtual Run VisitRun(Run node)
                         var value = node.Files[key];
                         if (value != null)
                         {
-                            node.Files[key] = VisitNullChecked(value);
+                            string newKey = key;
+                            node.Files.Remove(key);
+                            value = VisitNullChecked(value, ref newKey);
+
+                            if (newKey != null)
+                            {
+                                node.Files[newKey] = VisitNullChecked(value);
```

Comment:
>Files [](start = 37, length = 5)

Do you want a check for a duplicate key? Then maybe raise a warning to the user?